controllers/nodefeaturediscovery_status.go

```diff
@@ -0,0 +1,525 @@
+	// conditions should be removed.
+	for _, newCondition := range nfdCopy.Status.Conditions {
+		oldCondition := meta.FindStatusCondition(nfd.Status.Conditions, newCondition.Type)
+		if oldCondition == nil {
+			modified = true
+			break
+		}
+		// Ignore timestamps to avoid infinite reconcile loops
+		if oldCondition.Status != newCondition.Status ||
+			oldCondition.Reason != newCondition.Reason ||
+			oldCondition.Message != newCondition.Message {
+			modified = true
+			break
+		}
+	}
+
+	// If nothing has been modified, then return nothing. Even if the list
+	// of 'conditions' is not empty, it should not be counted as an update
+	// if it was already counted as an update before.
+	if !modified {
+		return nil
+	}
+	return r.Status().Update(context.TODO(), nfdCopy)
+}
+
+// updateDegradedCondition is used to mark a given resource as "degraded" so that
+// the reconciler can take steps to rectify the situation.
+func (r *NodeFeatureDiscoveryReconciler) updateDegradedCondition(nfd *nfdv1.NodeFeatureDiscovery, reason, message string) (ctrl.Result, error) {
+	degradedCondition := r.getDegradedConditions(reason, message)
+	if err := r.updateStatus(nfd, degradedCondition); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{Requeue: true}, nil
+}
+
+// updateProgressingCondition is used to mark a given resource as "progressing" so
+// that the reconciler can take steps to rectify the situation.
+func (r *NodeFeatureDiscoveryReconciler) updateProgressingCondition(nfd *nfdv1.NodeFeatureDiscovery, reason, message string) (ctrl.Result, error) {
+	progressingCondition := r.getProgressingConditions(reason, message)
+	if err := r.updateStatus(nfd, progressingCondition); err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{Requeue: true}, nil
+}
+
+// getAvailableConditions returns a list of conditionsv1.Condition objects and marks
+// every condition as FALSE except for conditionsv1.ConditionAvailable so that the
+// reconciler can determine that the resource is available.
+func (r *NodeFeatureDiscoveryReconciler) getAvailableConditions() []metav1.Condition {
+	now := time.Now()
+	return []metav1.Condition{
+		{
+			Type:               ConditionAvailable,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionUpgradeable,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionProgressing,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionDegraded,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+	}
+}
+
+// getDegradedConditions returns a list of conditions.Condition objects and marks
+// every condition as FALSE except for conditions.ConditionDegraded so that the
```

Comment:
```suggestion
// getDegradedConditions returns a list of Condition objects and marks
// every condition as FALSE except for ConditionDegraded so that the
```